server/api/src/rsGenQuery.cpp

```diff
@@ -405,39 +424,38 @@ irods::error strip_resc_hier_name_from_query_inp( genQueryInp_t* _inp, int& _pos
     // =-=-=-=-=-=-=-
     // cache pointers to the incoming inxIvalPair
     inxValPair_t tmpV;
+    const auto cleanup = irods::at_scope_exit{[&tmpV] { clearInxVal(&tmpV); }};
     tmpV.len   = _inp->sqlCondInp.len;
     tmpV.inx   = _inp->sqlCondInp.inx;
     tmpV.value = _inp->sqlCondInp.value;
 
+    const auto inxs = std::span{tmpV.inx, static_cast<std::size_t>(tmpV.len)};
+    const auto vals = std::span{tmpV.value, static_cast<std::size_t>(tmpV.len)};
+
     // =-=-=-=-=-=-=-
     // zero out the selectInp to copy
     // fresh indices and values
     std::memset(&_inp->sqlCondInp, 0, sizeof(_inp->selectInp));
 
-    // =-=-=-=-=-=-=-
-    // iterate over tmp and replace resource group with resource name
-    for ( int i = 0; i < tmpV.len; ++i ) {
-        if ( tmpV.inx[i] == COL_D_RESC_HIER ) {
-            std::string new_cond = "='0'";
-            irods::error ret = get_resc_id_cond_for_hier_cond(
-                                   tmpV.value[i],
-                                   new_cond);
-            if(!ret.ok()) {
-                irods::log(PASS(ret));
+    try {
+        // iterate over tmp and replace resource group with resource name
+        for (int i = 0; i < tmpV.len; ++i) {
+            const int inx = inxs[i];
+            const char* val = vals[i];
+            if (inx == COL_D_RESC_HIER) {
+                const auto new_cond = get_resc_id_cond_for_hier_cond(val);
+                addInxVal(&_inp->sqlCondInp, COL_D_RESC_ID, new_cond.empty() ? "='0'" : new_cond.c_str());
```

Comment:
Can `new_cond.c_str()` result in a dangling pointer when `new_cond` is destructed?

Reply:
`addInxVal` makes a copy of the passed in string for `condInput` via `strdup`, so I don't think that is a concern here:
https://github.com/irods/irods/blob/ec93f905d05776e67870ff80a92aba4227b926bd/lib/core/src/rcMisc.cpp#L866